### PR TITLE
VSCode documentations

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/39399

### CODE REVIEW

**Comment by jochman** [comment]
@yaron-libman please review (or assign someone else) to review `docs/concepts/vscode-extension.md`.
Thanks!

**Comment by ShirleyDenkberg** [comment]
@Itay4 Doc review completed.

File: docs/concepts/vscode-extension.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+* ***XSOAR: Load Script/Integration***: Opens a side-panel to easily modify the integration configuration, inputs, and outputs.
+
+* ***XSOAR: Demisto-SDK Lint/Validate/Update Release Notes...***: Will run the [demisto-sdk](https://github.com/demisto/demisto-sdk/) commands.
+
+## Debugging
+
+Read the [Debugging using your IDE](../integrations/debugging#Debugging%20using%20your%20IDE) section.
+
+## launch.json configuration
```
Comment (Itay4) [inline docs/concepts/vscode-extension.md:50]:
```suggestion
### launch.json configuration
```

File: docs/concepts/vscode-extension.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            "console": "integratedTerminal",
+            "cwd": "${workspaceFolder}" // workspaceFolder is the content repository
+        }
+    ]
+}
+```
+
+You can use Run and Debug (⇧⌘D) in VSCode (or click F5 on the python file).
+
+## Non-regular debugging
```
Comment (Itay4) [inline docs/concepts/vscode-extension.md:72]:
```suggestion
### Advanced Debugging
```

File: docs/integrations/debugging.md [hunk 3]
```diff
@@ -38,50 +39,75 @@ demisto.debug("I shouldn't have gone into STEM")
...
-```python 
+## Debugging using your IDE
+
+Sometimes when printing or using the logs is too confusing or messy you want to just use the debugger and go through the code line-by-line or breakpoint-by-breakpoint.
+
+> It is recommended to use the [Cortex XSOAR Visual Studio Code Extension](../concepts/vscode-extension.md) when you are developing content.
+
+### Python Environment  
+
+Python Environment
```
Comment (Itay4) [inline docs/integrations/debugging.md:57]: lets remove that section, it will confuse users with https://xsoar.pan.dev/docs/concepts/dev-setup

**Review by Itay4 — COMMENTED** [review]
looks good
1. please add the vscode ext article to the sidebar
2. we're missing a section about the right side menu for the integration settings

File: docs/concepts/vscode-extension.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+## Extension Workflow
+
+The extension defines a slightly different workflow than Cortex XSOAR. With the plugin, you can work on your code (whether it is Python, Powershell or Javascript).
+It is uses the excellent [demisto-sdk](./demisto-sdk) python package.
+
+## Prerequisites
+
+* Python 3.7 and up.
+* [Python for VSCode](https://marketplace.visualstudio.com/items?itemName=ms-python.python).
+* Install [demisto-sdk](../concepts/demisto-sdk#installation-and-setup). It is highly recommended to install it on a virtual environment such as [pipenv](https://pipenv.pypa.io/en/latest/).
```
[the remark this answers is not in the document]
Comment (jochman) [reply docs/concepts/vscode-extension.md:18]: We've changed the demisto-sdk to use pipenv (That's why we have now Pipenv)

**Comment by jochman** [comment]
@anara123 ? @glicht ?

**Comment by anara123** [comment]
@jochman you need to add the new page in the side bar. Under Developing.
And I think we should move the pycharm plugin article under `Articles`.
This way some users will still be able to find the article if they would like to, but it will under Developing side by side with VSCode which we recommend to use

**Comment by jochman** [comment]
> @jochman you need to add the new page in the side bar. Under Developing.
> And I think we should move the pycharm plugin article under `Articles`.
> This way some users will still be able to find the article if they would like to, but it will under Developing side by side with VSCode which we recommend to use
> 
> Also update the https://pull-request-763--demisto-content-docs.netlify.app/docs/integrations/unit-testing#setup-pycharm unit tests article, which has a section on pycharm plugin, then we need to add vscode part

We cant move them to articles as they're hardcoded in the CortexXSOAR platform.
I fixed the all other comments.

**Comment by anara123** [comment]
@jochman So maybe even better, keep the same id of the doc, create new doc for pycharm , and move the content of `Visual Studio Code Extension` into the existing doc. In the doc you will mention that we now recommend the new `Visual Studio Code Extension` instead of `PyCharm` and for those still insist on `PyCharm` can refer to doc X

**Review by anara123 — APPROVED** [review]

**Pull request merged**